stacktrace error when attempting to enable categories/categories & weighting (trunk-mysql)

When attempting to select the radio button for categories/categories & weighting in Settings, the following stacktrace gets generated:

`org.apache.wicket.WicketRuntimeException: Error attaching this container for rendering: [WebMarkupContainer [Component id = categoriesWrap]]
	at org.apache.wicket.MarkupContainer.onBeforeRenderChildren(MarkupContainer.java:1696)
	at org.apache.wicket.Component.onBeforeRender(Component.java:3826)
	at org.apache.wicket.Component.internalBeforeRender(Component.java:949)
	at org.apache.wicket.Component.beforeRender(Component.java:1017)
	at org.apache.wicket.Component.internalPrepareForRender(Component.java:2194)
	at org.apache.wicket.Component.prepareForRender(Component.java:2233)
	at org.apache.wicket.ajax.XmlAjaxResponse.writeComponent(XmlAjaxResponse.java:107)
	at org.apache.wicket.ajax.AbstractAjaxResponse.writeComponents(AbstractAjaxResponse.java:218)
	at org.apache.wicket.ajax.AbstractAjaxResponse.writeTo(AbstractAjaxResponse.java:150)
	at org.apache.wicket.ajax.AjaxRequestHandler.respond(AjaxRequestHandler.java:359)
	at org.apache.wicket.request.cycle.RequestCycle$HandlerExecutor.respond(RequestCycle.java:862)
	at org.apache.wicket.request.RequestHandlerStack.execute(RequestHandlerStack.java:64)
	at org.apache.wicket.request.RequestHandlerStack.execute(RequestHandlerStack.java:97)
	at org.apache.wicket.request.cycle.RequestCycle.execute(RequestCycle.java:261)
	at org.apache.wicket.request.cycle.RequestCycle.processRequest(RequestCycle.java:218)
	at org.apache.wicket.request.cycle.RequestCycle.processRequestAndDetach(RequestCycle.java:289)
	at org.apache.wicket.protocol.http.WicketFilter.processRequestCycle(WicketFilter.java:259)
	at org.apache.wicket.protocol.http.WicketFilter.processRequest(WicketFilter.java:201)
	at org.apache.wicket.protocol.http.WicketServlet.doPost(WicketServlet.java:159)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:648)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:729)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:292)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
	at org.sakaiproject.util.RequestFilter.doFilter(RequestFilter.java:411)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
	at org.apache.catalina.core.ApplicationDispatcher.invoke(ApplicationDispatcher.java:720)
	at org.apache.catalina.core.ApplicationDispatcher.processRequest(ApplicationDispatcher.java:466)
	at org.apache.catalina.core.ApplicationDispatcher.doForward(ApplicationDispatcher.java:358)
	at org.apache.catalina.core.ApplicationDispatcher.forward(ApplicationDispatcher.java:318)
	at org.sakaiproject.tool.impl.ActiveToolComponent$MyActiveTool.forward(ActiveToolComponent.java:513)
	at org.sakaiproject.portal.charon.SkinnableCharonPortal.forwardTool(SkinnableCharonPortal.java:1479)
	at org.sakaiproject.portal.charon.handlers.SiteHandler.doSite(SiteHandler.java:463)
	at org.sakaiproject.portal.charon.handlers.SiteHandler.doGet(SiteHandler.java:263)
	at org.sakaiproject.portal.charon.handlers.SiteHandler.doGet(SiteHandler.java:151)
	at org.sakaiproject.portal.charon.handlers.WorksiteHandler.doPost(WorksiteHandler.java:71)
	at org.sakaiproject.portal.charon.SkinnableCharonPortal.doPost(SkinnableCharonPortal.java:1157)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:648)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:729)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:292)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
	at org.sakaiproject.util.RequestFilter.doFilter(RequestFilter.java:460)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:212)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:106)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:502)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:141)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:79)
	at org.apache.catalina.valves.RemoteIpValve.invoke(RemoteIpValve.java:676)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:88)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:522)
	at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1095)
	at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:672)
	at org.apache.tomcat.util.net.AprEndpoint$SocketProcessor.doRun(AprEndpoint.java:2500)
	at org.apache.tomcat.util.net.AprEndpoint$SocketProcessor.run(AprEndpoint.java:2489)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NullPointerException
	at org.sakaiproject.gradebookng.tool.panels.SettingsCategoryPanel$8.populateItem(SettingsCategoryPanel.java:393)
	at org.apache.wicket.markup.html.list.ListView.onPopulate(ListView.java:523)
	at org.apache.wicket.markup.repeater.AbstractRepeater.onBeforeRender(AbstractRepeater.java:123)
	at org.apache.wicket.Component.internalBeforeRender(Component.java:949)
	at org.apache.wicket.Component.beforeRender(Component.java:1017)
	at org.apache.wicket.MarkupContainer.onBeforeRenderChildren(MarkupContainer.java:1684)
	... 62 more`

--- a/gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/SettingsCategoryPanel.java
+++ b/gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/SettingsCategoryPanel.java
@@ -13,6 +13,7 @@ import java.util.Locale;
 import java.util.Map;
 import java.util.Set;
 
+import org.apache.commons.lang.BooleanUtils;
 import org.apache.wicket.AttributeModifier;
 import org.apache.wicket.Component;
 import org.apache.wicket.ajax.AjaxEventBehavior;
@@ -318,6 +319,9 @@ public class SettingsCategoryPanel extends Panel {
 
 				final CategoryDefinition category = item.getModelObject();
 
+				// get the config. If there are no categories, detault is that the settings are enabled.
+				final boolean dropKeepEnabled = BooleanUtils.toBooleanDefaultIfNull(SettingsCategoryPanel.this.categoryDropKeepAvailability.get(category.getId()), true);
+
 				// note that all of these fields must have an ajaxform behaviour attached
 				// so that their data is persisted into the model.
 				// if they don't, when the listview repaints, they will be cleared
@@ -390,7 +394,8 @@ public class SettingsCategoryPanel extends Panel {
 						// intentionally left blank
 					}
 				});
-				categoryDropHighest.setEnabled(SettingsCategoryPanel.this.categoryDropKeepAvailability.get(category.getId()));
+				categoryDropHighest.setEnabled(dropKeepEnabled);
+
 				item.add(categoryDropHighest);
 
 				// drop lowest
@@ -405,7 +410,7 @@ public class SettingsCategoryPanel extends Panel {
 						// intentionally left blank
 					}
 				});
-				categoryDropLowest.setEnabled(SettingsCategoryPanel.this.categoryDropKeepAvailability.get(category.getId()));
+				categoryDropLowest.setEnabled(dropKeepEnabled);
 				item.add(categoryDropLowest);
 
 				// keep highest
@@ -420,7 +425,7 @@ public class SettingsCategoryPanel extends Panel {
 						// intentionally left blank
 					}
 				});
-				categoryKeepHighest.setEnabled(SettingsCategoryPanel.this.categoryDropKeepAvailability.get(category.getId()));
+				categoryKeepHighest.setEnabled(dropKeepEnabled);
 				item.add(categoryKeepHighest);
 
 				// remove button

